app/src/main/java/com/google/android/samples/socialite/repository/ChatRepository.kt

```diff
@@ -114,7 +118,17 @@ class ChatRepository @Inject internal constructor(
 
                 // Send a message prompt to the model to generate a response
                 var response = try {
-                    chat.sendMessage(text).text?.trim() ?: "..."
+                    if (mediaMimeType?.contains("image") == true) {
+                        val inputStream: InputStream? = appContext.contentResolver.openInputStream(
```

Comment:
What about using `appContext.contentResolver.openInputStream?.use`? This let you close input stream automatically.